### PR TITLE
feat(prettier): implement format function for ObjectPattern

### PR DESCRIPTION
**Pull request by Dunqing**
feat(prettier): implement format function for ObjectPattern

feat: replace space with Line

### CODE REVIEW

**Comment by Dunqing** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1303** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1303?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #1304** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1304?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #1305** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1305?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #1306** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1306?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #1307** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1307?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #1308** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1308?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #1309** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1309?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/oxc-project/oxc/1309?utm_source=stack-comment).

**Comment by Boshen** [comment]
### Merge activity

* **Nov 14, 7:22 AM**: @Boshen started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/oxc-project/oxc/1309).

File: crates/oxc_prettier/src/format.rs [hunk 1]
```diff
@@ -1232,7 +1232,20 @@ impl<'a> Format<'a> for BindingPattern<'a> {
 
 impl<'a> Format<'a> for ObjectPattern<'a> {
     fn format(&self, p: &mut Prettier<'a>) -> Doc<'a> {
-        Doc::Line
+        let mut parts = p.vec();
+        parts.push(p.str("{"));
+        parts.push(Doc::Line);
+        let max = self.properties.len() - 1;
```
[the remark this answers is not in the document]
Comment (Dunqing) [reply crates/oxc_prettier/src/format.rs:1238]: I don't think so. Nowhere is max used to get the item

[the remark this answers is not in the document]
Comment (Boshen) [reply crates/oxc_prettier/src/format.rs:1266]: It's always a good idea to be defensive ;-)

[the remark this answers is not in the document]
Comment (camc314) [reply crates/oxc_prettier/src/format.rs:1266]: `max` is a `usize`. if `self.properties.len()` is 0, then `0 - 1` will be -1 which will overflow

[the remark this answers is not in the document]
Comment (camc314) [reply crates/oxc_prettier/src/format.rs:1266]: meant overflow on the actual math method

[the remark this answers is not in the document]
Comment (Dunqing) [reply crates/oxc_prettier/src/format.rs:1266]: > `max` is a `usize`. if `self.properties.len()` is 0, then `0 - 1` will be -1 which will overflow

I see, Thank you! I will consider taking another way to fix it

**Pull request merged**